### PR TITLE
Swagger-ize the API

### PR DESCRIPTION
**Pull request by zelima**
This PR includes swagger documentations for CG API and some changes in API responses

After this PR it is available to see API documentation on `/api-docs` endpoint

PR includes 

- Small changes in serving API for reference data
  - responding with same structure as others `{total: ... results: [...]}` instead of directly rendering results
- Small changes in serving API for `/count` endpoint
  - responding with same structure as others `{total: ... results: [...]}`
- Tests modified according to changes above
- Serving of API removed from routes and moved to `api/controllers/...` dir. And the API is served by `swagger-express-mw` middle ware, with help of `swagger.yaml` file in controllers (magically)
- Main `app` swagger-ized in `app.js`
- serving `api/v1/swagger.json` for Swagger UI
- added 'api-docs' dir in `public` for rendering Swaager UI - this is copied and pasted from  here https://github.com/swagger-api/swagger-ui/tree/master/dist so no necessity to review anything in`api-docs` folder

### CODE REVIEW

**Comment by rufuspollock** [comment]
@zelima could you have @anuveyatsu do a first review of this PR. I can then check if needed.

**Comment by zelima** [comment]
@rufuspollock This includes mostly the visual changes All the functionalities remained same (just some of them moved into different file). The tests are passing and endpoints still returning same expected results.
Maybe will be easier to review when it's live after merged and deployed.
@anuveyatsu I would review one more time if there is any typos in `swagger.html` (if you have not already done) and would merge after

File: api/swagger/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+          required: false
+          type: string
+        - name: end
+          in: query
+          description: Exact ISO 8601 formatted date - Start date (inclusive)
+          required: false
+          type: string
+        - name: granularity
+          in: query
+          description: type of time granularity. options -  week, month, quarter, year.
```
Comment (anuveyatsu) [inline api/swagger/swagger.yaml]: Previously `description` started with a capital letter, so probably better to be consistent. Also, a new sentence should start with capital letter.

**Comment by zelima** [comment]
Merging

**Pull request merged**